src/org/openstreetmap/josm/actions/search/SearchAction.java

```diff
@@ -390,9 +452,14 @@ protected void buttonAction(int buttonIndex, ActionEvent evt) {
         if (dialog.showDialog().getValue() != 1) return null;
 
         // User pressed OK - let's perform the search
-        SearchMode mode = replace.isSelected() ? SearchAction.SearchMode.replace
-                : (add.isSelected() ? SearchAction.SearchMode.add
-                        : (remove.isSelected() ? SearchAction.SearchMode.remove : SearchAction.SearchMode.in_selection));
+        SearchMode mode = replace.isSelected()
```

Comment:
A sequence of chained ternary operators is from the original code. Will change to if/else then.